modules/codegen/src/main/scala/com/twilio/guardrail/generators/JavaGenerator.scala

```diff
@@ -0,0 +1,295 @@
+    }
+
+    def apply[T](term: ScalaTerm[JavaLanguage, T]): Target[T] = term match {
+      case LitString(value)        => Target.pure(new StringLiteralExpr(value))
+      case LitFloat(value)         => Target.pure(new DoubleLiteralExpr(value))
+      case LitDouble(value)        => Target.pure(new DoubleLiteralExpr(value))
+      case LitInt(value)           => Target.pure(new IntegerLiteralExpr(value))
+      case LitLong(value)          => Target.pure(new LongLiteralExpr(value))
+      case LitBoolean(value)       => Target.pure(new BooleanLiteralExpr(value))
+      case LiftOptionalType(value) => safeParseClassOrInterfaceType(s"java.util.Optional").map(_.setTypeArguments(new NodeList(value)))
+      case LiftOptionalTerm(value) => buildMethodCall("java.util.Optional.ofNullable", Some(value))
+      case EmptyOptionalTerm()     => buildMethodCall("java.util.Optional.empty")
+      case LiftVectorType(value)   => safeParseClassOrInterfaceType("java.util.List").map(_.setTypeArguments(new NodeList(value)))
+      case LiftVectorTerm(value)   => buildMethodCall("java.util.Collections.singletonList", Some(value))
+      case LiftMapType(value)      => safeParseClassOrInterfaceType("java.util.Map").map(_.setTypeArguments(STRING_TYPE, value))
+      case LookupEnumDefaultValue(tpe, defaultValue, values) => {
+        // FIXME: Is there a better way to do this? There's a gap of coverage here
+        defaultValue match {
+          case s: StringLiteralExpr =>
+            values
+              .find(_._1 == s.getValue)
+              .fold(Target.raiseError[Name](s"Enumeration ${tpe} is not defined for default value ${s.getValue}"))(value => Target.pure(value._3))
+          case _ =>
+            Target.raiseError(s"Enumeration ${tpe} somehow has a default value that isn't a string")
+        }
+      }
+      case EmbedArray(tpe) =>
+        tpe match {
+          case SwaggerUtil.Deferred(tpe) =>
+            Target.pure(SwaggerUtil.DeferredArray(tpe))
+          case SwaggerUtil.DeferredArray(_) =>
+            Target.raiseError("FIXME: Got an Array of Arrays, currently not supported")
+          case SwaggerUtil.DeferredMap(_) =>
+            Target.raiseError("FIXME: Got an Array of Maps, currently not supported")
+        }
+      case EmbedMap(tpe) =>
+        tpe match {
+          case SwaggerUtil.Deferred(inner) => Target.pure(SwaggerUtil.DeferredMap(inner))
+          case SwaggerUtil.DeferredMap(_) =>
+            Target.raiseError("FIXME: Got a map of maps, currently not supported")
+          case SwaggerUtil.DeferredArray(_) =>
+            Target.raiseError("FIXME: Got a map of arrays, currently not supported")
+        }
+      case ParseType(tpe) =>
+        safeParseType(tpe)
+          .map(Option.apply)
+          .recover {
+            case err =>
+              println(s"Warning: Unparsable x-java-type: ${tpe} ${err}")
```

Comment:
It seems like no split was actually done to make the java stuff use its own prefix; so long as it works, I've got no objections, so until `x-scala-type` -> `x-guardrail-type` or `x-guardrail-$lang-type`, I think we should just leave it `x-scala-type` to reduce confusion

Reply:
Yeah, haven't gotten around to that.  My thought was to split into `x-jvm-type`, `x-scala-type`, and `x-java-type`, since stuff like `AccountSid` will be applicable to java and scala.

Reply:
not a bad idea. I'm wondering if we should `x-guardrail` prefix these anyway, as to reduce the potential for name collisions

Reply:
either way, the keys would be able to fall back indefinitely, so I'm fine with whatever for now